x/oracle/keeper/keeper.go

```diff
@@ -113,10 +113,19 @@ func NewKeeper(
 
 // RemoveCurrencyPair removes a given CurrencyPair from state, i.e. removes its nonce + QuotePrice from the module's store.
 func (k *Keeper) RemoveCurrencyPair(ctx sdk.Context, cp slinkytypes.CurrencyPair) error {
+	// check if the currency pair exists.
+	if !k.HasCurrencyPair(ctx, cp) {
+		return types.NewCurrencyPairNotExistError(cp)
+	}
+
 	if err := k.currencyPairs.Remove(ctx, cp.String()); err != nil {
```

Comment:
this is just for knowledge sake - if this does not return an error if the key does not exist, what are the cases where this _does_ error?